### ISSUE TITLE
JVMTI SuspendResume1 test fails with invalid thread state

### ISSUE THREAD

**Issue by dipak-bagadiya** [eclipse-openj9/openj9#16690]
### Issue
JVMTI `GetThreadState` returns 0 (invalid state) for the carrier thread when a virtual thread is mounted. 0 refers to a thread which has not started but this state is invalid for the carrier thread. In `jvmtiHelpers.c::getThreadState`, `getVMThreadObjectStatesAll` returns `J9VMTHREAD_STATE_UNKNOWN` for the carrier thread. This will require further investigation.

### JVMTI Spec

GetThreadState: https://download.java.net/java/early_access/jdk19/docs/specs/jvmti.html#GetThreadState

### Test CMD
```
make test TEST="jtreg:test/hotspot/jtreg/serviceability/jvmti/vthread/SuspendResume1" JTREG="JAVA_OPTIONS=--enable-preview -Dvm.continuations=true;VERBOSE=all"
```

### Test Output
```
## Agent: thread[7] 0x7f0a4000b208 ForkJoinPool-1-worker-8: state after suspend: <none> (0)
## Agent: FAILED: SuspendThread did not turn on SUSPENDED flag for carrier thread:
#  state: <none> (0)
## Agent: SuspendThread:  virtual thread of carrier thread has state:  ALIVE RUNNABLE SUSPENDED (1048581)
STDERR:

Fatal error: check_resumed_state: expected SUSPENDED flag in thread state.
```

**Comment by babsingh** [comment]
Duplicate of https://github.com/eclipse-openj9/openj9/issues/15939. @fengxue-IS Can you point us to the PR which contains the fix for this issue?

**Comment by fengxue-IS** [comment]
see https://github.com/eclipse-openj9/openj9/pull/16374

**Comment by babsingh** [comment]
@dipak-bagadiya Can you check if changes from https://github.com/eclipse-openj9/openj9/pull/16374 are included in your local workspace? If not, fetch these changes and verify if the reported failure is fixed.

**Comment by dipak-bagadiya** [comment]
After #16374's change, the SuspendResume1 test no longer failed with ** ASSERTION FAILED ** at swalk.c, but still it returned the following error.

```
## Agent: thread[7] 0x7f0a4000b208 ForkJoinPool-1-worker-8: state after suspend: <none> (0)
## Agent: FAILED: SuspendThread did not turn on SUSPENDED flag for carrier thread:
#  state: <none> (0)
## Agent: SuspendThread:  virtual thread of carrier thread has state:  ALIVE RUNNABLE SUSPENDED (1048581)
STDERR:

Fatal error: check_resumed_state: expected SUSPENDED flag in thread state.
```

The aforementioned problem/error doesn't occur all the time.

**Comment by vij-singh** [comment]
What's the next step with this issue? (and who should own it?)

**Comment by babsingh** [comment]
> What's the next step with this issue? (and who should own it?)

VM team owns it. Updated issue's description and labels (fyi @dipak-bagadiya). The `Invalid JIT return address` issue was resolved with https://github.com/eclipse-openj9/openj9/pull/16374.

**Comment by babsingh** [comment]
Closing. Fix backported to the 0.37 release branch: https://github.com/eclipse-openj9/openj9/pull/16843.
